### ISSUE TITLE
How to train a general model for all temperature distribution in thermo-mechanical PDE?

### ISSUE THREAD

**Issue by mlin26** [lululxvi/deepxde#482]
Hi Dr. Lu,

I'd like a train a model that can get the results for all possible thermal distributions. The stress equation is as follows (I omit other equations just for simplicity):

![image](https://user-images.githubusercontent.com/48300409/149400357-a959f457-cf4a-471d-9d43-c6f88ef2c36a.png)

The 2D model size is 1 by 1. The temperature distribution dT here can be any temperature range from 0~1. The boundary conditions are the same. 

My naive thought is to sample all possible dT distributions and then randomly choose some of them to train. The dT can be treated as an input variable in x and stress as u in pde(x,u).

Could you tell me if DeepXDE has related module to implement this? Thanks!

**Comment by lululxvi** [comment]
Yes, it works. Just treat dT as another coordinate like x y z.

**Comment by mlin26** [comment]
> 

Dr. Lu, thanks for your response. It works now!

Another quick question... Could you tell me how to resample **anchors** data points for every given number of epochs:
`dde.data.PDE(geom, pde, [bc], num_domain=0, num_boundary=0, anchors=X)`

For example, I have 10000 data points and I only want to use random 1000 points for every given number of epochs. I have looked at '1D Possion resample' example, but it might not fit my needs. Thanks!

**Comment by lululxvi** [comment]
You might check [PDE.replace_with_anchors](https://deepxde.readthedocs.io/en/latest/modules/deepxde.data.html#deepxde.data.pde.PDE.replace_with_anchors)

**Comment by mlin26** [comment]
Dr. Lu, thanks so much for your help!

I have another question. When I train the model using both Adam and LBFGS, the training process would automatically stop with "Message: CONVERGENCE: REL_REDUCTION_OF_F_<=_FACTR*EPSMCH".

The console message is:
...........................................................................
INFO:tensorflow:Optimization terminated with:
  Message: CONVERGENCE: REL_REDUCTION_OF_F_<=_FACTR*EPSMCH
  Objective function value: 0.000424
  Number of iterations: 2
  Number of functions evaluations: 13
25853     [3.09e-05, 3.64e-05, 8.37e-05, 4.43e-05, 2.25e-04, 1.39e-06, 7.31e-07, 6.60e-07, 1.08e-06, 5.64e-09, 2.08e-08]    [3.09e-05, 3.64e-05, 8.37e-05, 4.43e-05, 2.25e-04, 1.39e-06, 7.31e-07, 6.60e-07, 1.08e-06, 5.64e-09, 2.08e-08]    []  

Best model at step 25853:
  train loss: 4.24e-04
  test loss: 4.24e-04
  test metric: []
...............................................................................

And sometimes LBFGS would automatically stop after 15000 epochs.  Could you tell me how I can let the optimizer not stop that early, for example, how to stop only after train loss < 1e-6?

My optimizer code is: 
` model.compile("adam", lr=0.001) `
`model.train(epochs=40000) `
`model.compile("L-BFGS") `
`losshistory, train_state = model.train()`

Thanks!

**Comment by lululxvi** [comment]
See https://deepxde.readthedocs.io/en/latest/modules/deepxde.optimizers.html#deepxde.optimizers.config.set_LBFGS_options
